Kernel/Heap/kmalloc.h

```diff
@@ -12,13 +12,29 @@
 #define KMALLOC_SCRUB_BYTE 0xbb
 #define KFREE_SCRUB_BYTE 0xaa
 
-#define MAKE_ALIGNED_ALLOCATED(type, alignment)                                                            \
-public:                                                                                                    \
-    [[nodiscard]] void* operator new(size_t) noexcept { return kmalloc_aligned<alignment>(sizeof(type)); } \
-    void operator delete(void* ptr) noexcept { kfree_aligned(ptr); }                                       \
-                                                                                                           \
+#define MAKE_ALIGNED_ALLOCATED(type, alignment)                                                                                   \
+public:                                                                                                                           \
+    [[nodiscard]] void* operator new(size_t)                                                                                      \
+    {                                                                                                                             \
+        void* ptr = kmalloc_aligned<alignment>(sizeof(type));                                                                     \
+        VERIFY(ptr);                                                                                                              \
+        return ptr;                                                                                                               \
+    }                                                                                                                             \
+    [[nodiscard]] void* operator new(size_t, const std::nothrow_t&) noexcept { return kmalloc_aligned<alignment>(sizeof(type)); } \
+    void operator delete(void* ptr) noexcept { kfree_aligned(ptr); }                                                              \
+                                                                                                                                  \
 private:
 
+// The C++ standard specifies that the nothrow allocation tag should live in the std namespace.
+// Otherwise, `new (std::nothrow)` calls wouldn't get resolved.
+namespace std {
+struct nothrow_t {
+    explicit nothrow_t() = default;
+};
+
+extern const nothrow_t nothrow;
+};
+
```

Comment:
* Is this the right header (Types.h or a LibC header seem more appropriate)?
* May compilers export this on their own into the std namespace
* Can you move the `using std::nothrow_t` to the definition
* please omit the `std::` in all the references to it

Reply:
> Is this the right header (Types.h or a LibC header seem more appropriate)?
> May compilers export this on their own into the std namespace?

We definitely should not put C++-specific types in LibC, so that is out of the question.
For the userland and ports, this `std::nothrow_t` is defined by the `<new>` standard library header, which we explicitly do not include in any kernel code (see `AK/kmalloc.h`).  Therefore, this definition must be local to the kernel, and since it's only used for `kmalloc`, it makes sense to group this with it.

> May compilers export this on their own into the std namespace?

AFAIK, no compilers will include the `<new>` standard library header unless we explicitly ask for it. So if you are asking about potential name collisions, I don't think that can happen.

> Can you move the using std::nothrow_t to the definition
> please omit the std:: in all the references to it

Note that we only export the `std::nothrow` constant, not its `std::nothrow_t` type. I think this case falls under the same umbrella as `std::nullptr_t`, which we also always use by its fully qualified name.

I am open to more in-depth discussion or clarifications about this, but I think it would be more productive to discuss it on Discord.